src/read_dta.cpp

```diff
@@ -579,11 +579,11 @@ List read_dta(FILE * file, const bool missing, const IntegerVector selectrows) {
 
         // works for LSF on little- and big-endian
         if(byteorder.compare("LSF")==0) {
-          v = (int32_t)z;
+          v = (int32_t)z & ((1 << 24) - 1);
```

Comment:
This works. Did check it with additional data.frame with 2 variables. I'm still surprised it does, though.